resources/js/Components/LeftMenu.tsx

```diff
@@ -1,11 +1,70 @@
+import axios from "axios";
 import Brand from "./Brand";
 import {
     HomeIcon,
     ArchiveBoxIcon,
     BookOpenIcon,
 } from "@heroicons/react/24/solid";
+import useSWR from "swr";
+
+interface CategoryParameters {
+    categoryName: string;
+    linksArray: Array<LinksArrayParameters>;
+    rank: number;
+}
+interface LinksArrayParameters {
+    [linkName: string]: string;
+}
+
+function Category({ categoryName, linksArray, rank }: CategoryParameters) {
+    const linksList = linksArray.map((linkPair: { [x: string]: string }) => {
+        const key = Object.keys(linkPair)[0];
+        return (
+            <li key={key.concat(rank.toString())}>
+                <a href={linkPair[key]}>{key}</a>
+            </li>
+        );
+    });
+    return (
+        <li>
+            <details>
+                <summary>
+                    <ArchiveBoxIcon className="w-4" />
+                    {categoryName}
+                </summary>
+                <ul>{linksList}</ul>
+            </details>
+        </li>
+    );
+}
+
+function getCategoryItems() {
+    const { data, error, isLoading } = useSWR("/api/v1/categories", (url) =>
```

Comment:
Technically, all hooks need to be in the body of the component and shouldn't be placed in functions. I would moev useSWR into the LeftMenu component and pass the data into this function.